pickup/lib/s3.js

```diff
@@ -87,16 +73,24 @@ export class S3Uploader {
     /**
      * @typedef {import('node:stream').Readable} Readable
      * @param {Readable} body
+     * @param {AbortSignal} signal
      */
-    return async function (body) {
-      return uploadAndVerify({
+    return async function (body, signal) {
+      // Handles s3 multipart uploading
+      // see: https://github.com/aws/aws-sdk-js-v3/blob/main/lib/lib-storage/README.md
+      const s3Upload = new Upload({
         client: s3,
-        validationBucket,
-        destinationBucket,
-        key,
-        body,
-        cid
+        params: {
+          Metadata: { structure: 'complete' },
+          Bucket: validationBucket,
+          Key: key,
+          Body: body
+        }
       })
+      signal.addEventListener('abort', () => s3Upload.abort())
```

Comment:
important line! can't pass an abort signal to the s3Uploader oh no. but you can call abort in it so we deal with that here, to ensure the stream is destroyed when the abort signal is fired by our fetch an chunk timeouts.